File: src/modules/orderbook/dtos/order-response.dto.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { OrderSide, OrderStatus, OrderStatusActor } from '../enums/order.enum';

export class OrderStatusHistoryResponseDto {
  @ApiProperty()
  status: OrderStatus;

  @ApiProperty()
  actor: OrderStatusActor;

  @ApiProperty()
  created_at: string;
}
export class OrderResponseDto {
  @ApiProperty()
  id: number;

  @ApiProperty({ enum: OrderSide })
  side: OrderSide;

  @ApiProperty()
  price: string;

  @ApiProperty()
  quantity: string;

  @ApiProperty()
  remaining: string;

  @ApiProperty({ enum: OrderStatus })
  status: OrderStatus;

  @ApiProperty()
  created_at: string;

  @ApiProperty()
  updated_at: string;

  @ApiPropertyOptional()
  user_id: number;

  @ApiPropertyOptional()
  status_history?: OrderStatusHistoryResponseDto[];
}

export class OrderbookLevelDto {
  @ApiProperty()
  price: string;

  @ApiProperty()
  remaining: number;

  @ApiProperty()
  user_name: string;

  @ApiProperty()
  status: OrderStatus;

  @ApiProperty()
  side: OrderSide;

  @ApiProperty()
  created_at: string;
}

File: src/main/java/com/sideprojects/tradematching/dto/order/OrderResponseDto.java
package com.sideprojects.tradematching.dto.order;

import com.sideprojects.tradematching.entity.enums.OrderSide;
import com.sideprojects.tradematching.entity.enums.OrderStatus;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponseDto {

    private Long id;

    @JsonProperty("user_id")
    private Long userId;

    private OrderSide side;
    private String price;
    private String quantity;
    private String remaining;
    private OrderStatus status;

    @JsonProperty("validity_days")
    private Integer validityDays;

    @JsonProperty("created_at")
    private String createdAt;

    @JsonProperty("updated_at")
    private String updatedAt;

    @JsonProperty("status_history")
    private List<OrderStatusHistoryResponseDto> statusHistory;
}
